src/main/kotlin/net/maizegenetics/phgv2/cli/AlignAssemblies.kt

```diff
@@ -0,0 +1,320 @@
+            throw IllegalStateException("Error running minimap2 for reference: $error")
+        }
+
+        runAnchorWaveMultiThread(ref, assembliesList, cdsFasta, gff, refSamOutFile)
+
+    }
+
+    private fun createCDSfromRefData(refFasta: String, gffFile: String, cdsFasta: String, outputDir: String): Boolean {
+
+        // val command = "anchorwave gff2seq -r ${refFasta} -i ${gffFile} -o ${cdsFasta} "
+        // Need to set the conda environment here to access anchorwave
+        val builder = ProcessBuilder(
+            "conda",
+            "run",
+            "-n",
+            "phgv2-conda",
+            "anchorwave",
+            "gff2seq",
+            "-r",
+            refFasta,
+            "-i",
+            gffFile,
+            "-o",
+            cdsFasta
+        )
+
+        val redirectOutput = "$outputDir/anchorwave_gff2seq_output.log"
+        val redirectError = "$outputDir/anchorwave_gff2seq_error.log"
+        builder.redirectOutput(File(redirectOutput))
+        builder.redirectError(File(redirectError))
+
+        println("createCDSfromRefData command:" + builder.command().stream().collect(Collectors.joining(" ")))
+        try {
+            val process = builder.start()
+            val error = process.waitFor()
+            if (error != 0) {
+                println("createCDSfromRefData run via ProcessBuilder returned error code $error")
+                return false
+            }
+            return true
+        } catch (e: Exception) {
+            println("Error: could not execute anchorwave command. Run anchorwave manually and retry.")
+            return false
+        }
+
+    }
+
+    private fun runAnchorWaveMultiThread(
+        refFasta: String,
+        assemblies: List<String>,
+        cdsFasta: String,
+        gffFile: String,
+        refSamOutFile: String
+    ) {
+        runBlocking {
+            // Setup
+            val inputChannel = Channel<InputChannelData>()
+
+            // The input channel gets data needed to run minimap2 and align with anchorwave
+            launch {
+                println("Adding entries to the inputChannel:")
+                assemblies.forEach { asmFile ->
+
+                    // Column names were checked for validity above
+
+                    println("Adding: $asmFile for processing")
+                    inputChannel.send(InputChannelData(refFasta, asmFile, outputDir, gffFile, refSamOutFile))
+                }
+                println("Done Adding data to the inputChannel:")
+                inputChannel.close() // Need to close this here to show the workers that it is done adding more data
+            }
+
+            // This calls mummer4 scripts to process the alignments
+            val workerThreads = (1..runs).map { run ->
+                launch { alignAssembly(inputChannel, cdsFasta, gffFile) }
+            }
+
+            // Do not need a coroutine that "joins" the threads as they will all
```

Comment:
Maybe move this comment up a few lines above the val workerThreads....  As it is now it looks like you are missing code at the bottom of the method at a glance.

Reply:
Moved